crates/signer/src/wallet/private_key.rs

```diff
@@ -0,0 +1,406 @@
+//! Specific helper functions for loading an offline K256 Private Key stored on disk
+
+use super::{Wallet, WalletError};
+use crate::utils::secret_key_to_address;
+use alloy_primitives::hex;
+use k256::{
+    ecdsa::{self, SigningKey},
+    FieldBytes, SecretKey as K256SecretKey,
+};
+use rand::{CryptoRng, Rng};
+use std::str::FromStr;
+
+#[cfg(feature = "keystore")]
+use {elliptic_curve::rand_core, std::path::Path};
+
+impl Wallet<SigningKey> {
+    /// Creates a new Wallet instance from a raw scalar serialized as a byte array.
+    #[inline]
+    pub fn from_bytes(bytes: &FieldBytes) -> Result<Self, ecdsa::Error> {
+        SigningKey::from_bytes(bytes).map(Self::new_pk)
+    }
+
+    /// Creates a new Wallet instance from a raw scalar serialized as a byte slice.
+    #[inline]
+    pub fn from_slice(bytes: &[u8]) -> Result<Self, ecdsa::Error> {
+        SigningKey::from_slice(bytes).map(Self::new_pk)
+    }
+
+    /// Creates a new random keypair seeded with [`rand::thread_rng()`].
+    #[inline]
+    pub fn random() -> Self {
+        Self::random_with(&mut rand::thread_rng())
+    }
+
+    /// Creates a new random keypair seeded with the provided RNG.
+    #[inline]
+    pub fn random_with<R: Rng + CryptoRng>(rng: &mut R) -> Self {
```

Comment:
can these be feature-gated? Generally we don't want people to be doing this 😅

Reply:
Why not? Rand is brought in from deps anyway, and it's nice in tests

Reply:
Creating a random raw key is essentially never best practice, and shouldn't be in the standard API. We want to recommend users not use this API and feature gating it is a good way to do that